components/collector/src/source_collectors/quality_time/missing_metrics.py

```diff
@@ -83,7 +83,7 @@ def __subject_missing_metric_type_entities(
                 key=f"{report_uuid}:{subject_uuid}:{metric_type}",
                 report=report["title"],
                 report_url=report_url,
-                subject=subject.get("name", subject_type_name),
+                subject=subject.get("name") or subject_type_name,
```

Comment:
Is this not something we can address in the datamodel, or does the issue derive from only storing stringy data in the db?

Reply:
In theory this could fixed by making sure domain objects always have all attributes present (even if with an empty string or empty list as value). Downside is the extra storage needed. Impact could be considerable if all measurement entities have all their attributes (entity status, status rationale, status end date, etc.) stored at all times. Especially when we remove the limit of storing 100 measurement entities. Maybe Mongo deals with the duplication, but it would require some looking into.